### PR TITLE
Archive labels UI enhancement

### PR DESCRIPTION
**Pull request by puni9869**
Followup  https://github.com/go-gitea/gitea/pull/26478
## Archived labels UI

Added:

* Implemented the structural framework for the user interface to accommodate archived labels.
* Introduced archived labels functionality to the UI, akin to the approach taken for closed milestones or projects. This functionality allows concealing information while retaining it, without resorting to deletion.
* Particularly advantageous for managing outdated labels that are no longer in use, offering a means to preserve them without cluttering the active label set.

Changed:
* Enhanced the Filtered UI page to seamlessly incorporate a list of archived labels.

Outsourced:
* Defer the implementation of specialized handling for archived labels to upcoming pull requests. This step will be undertaken subsequent to the successful merge of this pull request.

<details><summary>Screenshots</summary>

image

image

image

</details>

---

Part of https://github.com/go-gitea/gitea/issues/25237

### CODE REVIEW

File: templates/repo/issue/labels/label_list.tmpl [hunk 1]
```diff
@@ -30,7 +38,7 @@
 
 	<ul class="issue-label-list">
 		{{range .Labels}}
-		<li class="item">
+		<li class="item" {{if gt .ArchivedUnix 0}}data-archived-label{{end}}>
```
Comment (KN4CK3R) [inline templates/repo/issue/labels/label_list.tmpl]:
```suggestion
		<li class="item" {{if not .ArchivedUnix.IsZero}}data-archived-label{{end}}>
```

**Review by KN4CK3R — COMMENTED** [review]
What's the benefit of toggling the hightlight?

**Comment by puni9869** [comment]
> What's the benefit of toggling the hightlight?

Here is discussion
https://github.com/go-gitea/gitea/pull/26502#discussion_r1293997856

**Comment by KN4CK3R** [comment]
Ok, but I still think a static archived highlight is all what's needed.

**Comment by puni9869** [comment]
Introduced archived labels functionality to the UI, akin to the approach taken for closed milestones or projects. This functionality allows concealing information while retaining it, without resorting to deletion.

File: web_src/js/features/repo-issue-list.js [hunk 2]
```diff
@@ -203,3 +203,13 @@ export function initRepoIssueList() {
   initRepoIssueListAuthorDropdown();
   initIssuePinSort();
 }
+
+export function initHighlightArchivedLabels() {
+  if (!document.querySelectorAll('.show-archived-labels-checkbox').length) return;
+  document.querySelector('.show-archived-labels-checkbox')
```
Comment (delvh) [inline web_src/js/features/repo-issue-list.js]:
```suggestion
  if (!document.querySelectorAll('.show-archived-labels-checkbox').length) return;
  const checkbox = document.querySelector('.show-archived-labels-checkbox');
  if (!checkbox) return;
  checkbox.
```

**Comment by puni9869** [comment]
> I meant **hide** archived labels by default, and show them only when the checkbox is toggled.

I am going to change again on the above basis.

**Comment by puni9869** [comment]
image

I want to add highlight for initial release of this feature.
image

**Comment by KN4CK3R** [comment]
Why not just sort them to the bottom and add the highlight? Then you don't need extra logic to show them and they don't block the view.

**Comment by delvh** [comment]
Works too.
Except then we need the extra logic to sort them.

**Comment by puni9869** [comment]
> Why not just sort them to the bottom and add the highlight? Then you don't need extra logic to show them and they don't block the view.

I can managed that with js

**Comment by KN4CK3R** [comment]
> Works too. Except then we need the extra logic to sort them.

The extra logic is "sort by archived date"

**Comment by delvh** [comment]
`Sort primarily by archive date and then by name`.
True, that works.

**Comment by KN4CK3R** [comment]
To keep the archived labels sorted by name too you could do the sorting in Go or sort by `if archived date > 0 ? 1 : 0` in sql.

**Comment by puni9869** [comment]
image

**Comment by KN4CK3R** [comment]
Keep in mind that the archived labels are only sorted by name because the timestamps are in the correct order.

**Comment by puni9869** [comment]
> Keep in mind that the archived labels are only sorted by name because the timestamps are in the correct order.

Yes, correct, 
Query will be complex

**Comment by puni9869** [comment]
Let's scoped this PR for Ui changes only. I am sending a back to back Pr for Sorting and Db migration changes [We need to index the archived_unix] column and a separate method I am introducing because the older method is used several places.
Just to avoid the conflicts in result set.

Need approvals/reviews regarding UI changes only. 
@delvh  
cc/ @KN4CK3R

File: web_src/js/features/repo-issue-list.js [hunk 3]
```diff
@@ -203,3 +203,13 @@ export function initRepoIssueList() {
...
+    .addEventListener('input', () => {
```
Comment (delvh) [inline web_src/js/features/repo-issue-list.js]:
```suggestion
    .addEventListener('input', (e) => {
```

File: web_src/js/features/repo-issue-list.js [hunk 4]
```diff
@@ -203,3 +203,13 @@ export function initRepoIssueList() {
...
+      for (const archivedLabel of document.querySelectorAll('.archived-label')) {
+        archivedLabel.classList.toggle('gt-hidden');
```
Comment (delvh) [inline web_src/js/features/repo-issue-list.js]:
```suggestion
        toggleElem(archivedLabel, e.target.checked);
```

Comment (puni9869) [reply web_src/js/features/repo-issue-list.js]: `toggleElem(archivedLabel, e.target.checked);` I will call it Black Magic

**Review by delvh — APPROVED** [review]

**Comment by KN4CK3R** [comment]
If we sort them in the next PR, why is the toggle code still there?

**Comment by puni9869** [comment]
+1 more approval needed on this PR.

**Comment by puni9869** [comment]
> If we sort them in the next PR, why is the toggle code still there?

Your point perfectly valid. Currently I am working on the sort functionality.  I will mark this PR `reviewed-wait`. and I am about to raise the sort PR.

**Comment by delvh** [comment]
You mean WIP, right?

**Comment by puni9869** [comment]
> This yellow highlight bar looks very odd, is it removed in latest versions?

We can change the color. I took it from base.css

**Comment by silverwind** [comment]
What does it highlight? Archived state? I guess a small "Archived" label would be equally well suited, and less confusing. Maybe in its own column.

File: web_src/js/features/repo-issue-list.js [hunk 5]
```diff
@@ -203,3 +203,15 @@ export function initRepoIssueList() {
   initRepoIssueListAuthorDropdown();
   initIssuePinSort();
 }
+
+export function initHighlightArchivedLabels() {
+  if (!document.querySelectorAll('.show-archived-labels-checkbox').length) return;
+  const showArchivedLabelCheckBox = document.querySelector('.show-archived-labels-checkbox');
+  if (!showArchivedLabelCheckBox) return;
+  document.querySelector('.show-archived-labels-checkbox')
```
Comment (wxiaoguang) [inline web_src/js/features/repo-issue-list.js]: Querying the same selector 3 times?

image

File: templates/repo/issue/labels/label_list.tmpl [hunk 6]
```diff
@@ -1,6 +1,14 @@
-<h4 class="ui top attached header">
-	{{.locale.Tr "repo.issues.label_count" .NumLabels}}
-	<div class="ui right">
+<h4 class="ui top attached header gt-df">
+	<span class="gt-f1">{{.locale.Tr "repo.issues.label_count" .NumLabels}}</span>
+	<label class="gt-f1" data-tooltip-content={{.locale.Tr "repo.issues.label_archive_tooltip"}}>
```
Comment (wxiaoguang) [inline templates/repo/issue/labels/label_list.tmpl]: The position seems pretty strange. 

IMO it should be either left aligned or right aligned.

image

File: templates/repo/issue/labels/label_list.tmpl [hunk 7]
```diff
@@ -1,6 +1,14 @@
...
+		<input type="checkbox"
+			name="show_archived_labels"
+			class="show-archived-labels-checkbox"
+		>
+		{{.locale.Tr "repo.issues.show_archived_labels"}}
+	</label>
```
Comment (wxiaoguang) [inline templates/repo/issue/labels/label_list.tmpl:9]: It seems that Gitea is not ready for using native checkbox at the moment, due to the misalignment

image

Comment (puni9869) [reply templates/repo/issue/labels/label_list.tmpl:9]: Done

Comment (puni9869) [reply templates/repo/issue/labels/label_list.tmpl:9]: image

Comment (puni9869) [reply templates/repo/issue/labels/label_list.tmpl:9]: image

**Comment by puni9869** [comment]
Follup to https://github.com/go-gitea/gitea/pull/26741

**Comment by puni9869** [comment]
Closing as of now.

**Pull request closed without merging**